drv/transceivers-server/src/main.rs

```diff
@@ -125,70 +126,49 @@ impl idl::InOrderTransceiversImpl for ServerImpl {
             .map_err(TransceiversError::from)?)
     }
 
-    fn set_power_enable(
+    fn all_power_states(
         &mut self,
         _msg: &userlib::RecvMessage,
-        mask: u32,
-    ) -> Result<(), idol_runtime::RequestError<TransceiversError>> {
-        self.transceivers
-            .set_power_enable(mask)
-            .map_err(TransceiversError::from)?;
-        Ok(())
-    }
-
-    fn clear_power_enable(
-        &mut self,
-        _msg: &userlib::RecvMessage,
-        mask: u32,
-    ) -> Result<(), idol_runtime::RequestError<TransceiversError>> {
-        self.transceivers
-            .clear_power_enable(mask)
-            .map_err(TransceiversError::from)?;
-        Ok(())
-    }
-
-    fn set_reset(
-        &mut self,
-        _msg: &userlib::RecvMessage,
-        mask: u32,
-    ) -> Result<(), idol_runtime::RequestError<TransceiversError>> {
-        self.transceivers
-            .set_reset(mask)
-            .map_err(TransceiversError::from)?;
-        Ok(())
+    ) -> Result<PowerStatesAll, idol_runtime::RequestError<TransceiversError>>
+    {
+        Ok(self
+            .transceivers
+            .all_power_states()
+            .map_err(TransceiversError::from)?)
     }
 
-    fn clear_reset(
+    fn set_power_state(
         &mut self,
         _msg: &userlib::RecvMessage,
         mask: u32,
+        state: PowerState,
     ) -> Result<(), idol_runtime::RequestError<TransceiversError>> {
-        self.transceivers
-            .clear_reset(mask)
-            .map_err(TransceiversError::from)?;
-        Ok(())
+        Ok(self
```

Comment:
Style nit: I don't love returning `Ok(some_function()?)` in cases when `some_function` returns a `Result<(), ...>`, because it's less obvious that we're not returning anything.

(this also applies to the two functions below)